### PR TITLE
Configuration updates

### PR DESCRIPTION
**Pull request by jswift-stfc**
- Added configuration for SANDALS (2011).
- Renamed configuration for NIMROD (2012, 84 modules) to be more informative.
- Added descriptions to configuration files.
- Removed 'load configuration' menu option.
- Imposed loading of configurations through a dialog, whilst creating a 'new' input file.
- Configurations are shown in a `QListWidget` in the dialog, with the description shown beneath in a `QTextEdit`.

### CODE REVIEW

**Comment by trisyoungs** [comment]
Unfortunately, on choosing File->New:

```
Traceback (most recent call last):
  File "src\gui\widgets\main_window.py", line 555, in newInputFile
  File "src\gui\widgets\dialogs\configuration_dialog.py", line 29, in __init__
  File "src\gui\widgets\dialogs\configuration_dialog.py", line 52, in initComponents
AttributeError: 'PySide6.QtWidgets.QDialog' object has no attribute 'configList'
```

**Comment by jswift-stfc** [comment]
> Unfortunately, on choosing File->New:
> 
> ```
> Traceback (most recent call last):
>   File "src\gui\widgets\main_window.py", line 555, in newInputFile
>   File "src\gui\widgets\dialogs\configuration_dialog.py", line 29, in __init__
>   File "src\gui\widgets\dialogs\configuration_dialog.py", line 52, in initComponents
> AttributeError: 'PySide6.QtWidgets.QDialog' object has no attribute 'configList'
> ```

Typo in the executable version! Fixed now I believe.

File: src/gui/widgets/dialogs/configuration_dialog.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        super(ConfigurationDialog, self).__init__(parent=parent)
+        self.parent = parent
+        self.configuration = None
+        self.cancelled = False
+        self.initComponents()
+        self.loadConfigurations()
+
+    def initComponents(self):
+        """
+        Loads the UI file for the ViewInputDialog object.
```
Comment (trisyoungs) [inline src/gui/widgets/dialogs/configuration_dialog.py:34]:
```suggestion
        Loads the UI file for the ConfigurationDialog object.
```

**Review by trisyoungs — APPROVED** [review]
:+1:

**Pull request merged**